### ISSUE TITLE
Exclude/Include Folders

### ISSUE THREAD

**Issue by Eyenseo** [franciscofranco/Focus-bugtracker#25]
This is a bug / feature request.

Currently Focus shows all album covers from my external card that are in the music folder. This is very annoying since I have to scroll through 523 albums - that said I do not want to hide all 523 albums manually.

Please let me specify the folders to look for pictures in - currently the app is barely useful for me!

**Comment by franciscofranco** [comment]
Yeah we received that feedback a couple times. I just didn't implement something like this because I don't really know of a good solution both UI and code wise. Can you elaborate as how you would like to be presented with a functionality like this? I mean I can't just "figure" out which folders have pictures you don't want to be scanned.

**Comment by Eyenseo** [comment]
@franciscofranco For the GUI you might want to have a look at Poweramp. They present you with a collapsible tree view of your file system. At the right hand side of each item there is a checkbox to include a Folder recursively. Subfolders can be then again be deselected. Additionally to the checkbox the selected folders are color coded to display their include status.
Alternatively have a look at QuickPic, though I would prefer Poweramps way. 

Code wise: I assume the user selected the paths to the folders that should be included (by default you could just pre-select /sdcard/DCIM for example). The next step would be to generate a list of all folders (inclusive subfolders) that should be searched. From this list you have to remove the folders that are marked as exclude (that list should be created when a folder gets deselected -> a parent folder was selected). The result is a list of all folders that have to be searched and you can ignore the subfolders as they are already in the list.

I hope this gives you an idea what I would like to have ;)
